R-package/tests/testthat/test_lgb.Booster.R

```diff
@@ -240,6 +240,68 @@ test_that("Loading a Booster from a string works", {
     expect_identical(pred, pred2)
 })
 
+test_that("Saving a large model to string should work", {
+    set.seed(708L)
+    data(agaricus.train, package = "lightgbm")
+    train <- agaricus.train
+    bst <- lightgbm(
+        data = as.matrix(train$data)
+        , label = train$label
+        , num_leaves = 100L
+        , learning_rate = 0.01
+        , nrounds = 500L
+        , objective = "binary"
+        , save_name = tempfile(fileext = ".model")
+        , verbose = -1L
+    )
+
+    pred <- predict(bst, train$data)
```

Comment:
Nice test!
I believe that with `rawscore = TRUE` or `predleaf = TRUE` this test will become even more strict and reliable.

Reply:
updated in https://github.com/microsoft/LightGBM/pull/4256/commits/6430f62dfc683449f6ac01d51a67505c7947efbc